ipc/arwenpart/blockchainGateway.go

```diff
@@ -442,11 +437,85 @@ func (blockchain *BlockchainHookGateway) GetCompiledCode(codeHash []byte) (bool,
 	return response.Found, response.Code
 }
 
-// ClearCompiledCodes nothing to do - this needs to be called by nodepart only
+// ClearCompiledCodes forwards a message to the actual hook
 func (blockchain *BlockchainHookGateway) ClearCompiledCodes() {
+
+	request := common.NewMessageBlockchainClearCompiledCodesRequest()
+	rawResponse, err := blockchain.messenger.SendHookCallRequest(request)
+	if err != nil {
+		return
+	}
+
+	if rawResponse.GetKind() != common.BlockchainClearCompiledCodesResponse {
+		return
+	}
+
+	return
+}
+
+// GetESDTToken forwards a message to the actual hook
+func (blockchain *BlockchainHookGateway) GetESDTToken(address []byte, tokenID []byte, nonce uint64) (*esdt.ESDigitalToken, error) {
+
+	request := common.NewMessageBlockchainGetESDTTokenRequest(address, tokenID, nonce)
+	rawResponse, err := blockchain.messenger.SendHookCallRequest(request)
+	if err != nil {
+		return nil, err
+	}
+
+	if rawResponse.GetKind() != common.BlockchainGetESDTTokenResponse {
+		return nil, common.ErrBadHookResponseFromNode
+	}
+
+	response := rawResponse.(*common.MessageBlockchainGetESDTTokenResponse)
+	return response.Result, response.GetError()
 }
 
-// IsInterfaceNil returns true if underlying implementation is nil
+// IsInterfaceNil forwards a message to the actual hook
 func (blockchain *BlockchainHookGateway) IsInterfaceNil() bool {
-	return blockchain == nil
+
```

Comment:
this is wrong. There is no need for IPC messaging for IsInterfaceNil. the only code here should be return blockchain == nil